pkg/apis/apps/v1beta1/conversion.go

```diff
@@ -178,48 +180,35 @@ func Convert_apps_StatefulSetUpdateStrategy_To_v1beta1_StatefulSetUpdateStrategy
 	return nil
 }
 
-func Convert_extensions_ScaleStatus_To_v1beta1_ScaleStatus(in *extensions.ScaleStatus, out *appsv1beta1.ScaleStatus, s conversion.Scope) error {
+func Convert_autoscaling_ScaleStatus_To_v1beta1_ScaleStatus(in *autoscaling.ScaleStatus, out *appsv1beta1.ScaleStatus, s conversion.Scope) error {
 	out.Replicas = int32(in.Replicas)
+	out.TargetSelector = in.Selector
 
 	out.Selector = nil
-	out.TargetSelector = ""
-	if in.Selector != nil {
-		if in.Selector.MatchExpressions == nil || len(in.Selector.MatchExpressions) == 0 {
-			out.Selector = in.Selector.MatchLabels
-		}
-
-		selector, err := metav1.LabelSelectorAsSelector(in.Selector)
-		if err != nil {
-			return fmt.Errorf("invalid label selector: %v", err)
-		}
-		out.TargetSelector = selector.String()
+	selector, err := metav1.ParseToLabelSelector(in.Selector)
+	if err != nil {
+		return fmt.Errorf("failed to parse selector: %v", err)
 	}
+	if len(selector.MatchExpressions) == 0 {
```

Comment:
Do we need to handle the case when `len(selector.MatchExpressions) != 0`?

Reply:
no, if `MatchExpressions` is non-empty, then `MatchLabels` is insufficient to express the full selector, and we should not set it.

This is the same logic as before:
```
if in.Selector.MatchExpressions == nil || len(in.Selector.MatchExpressions) == 0 {
	out.Selector = in.Selector.MatchLabels
}
```

Reply:
Good point, thanks!